### DISCUSSION TITLE
How to get a normal "plagiotropous"  at the vertices of the two faces?

### DISCUSSION THREAD

**Discussion by sinkinthesea** [jrouwe/JoltPhysics#1312] [Q&A]
Emm, in these pictures, normal is different in diffrent physics.
I used Api : ShapeCast with a CapsuleShape.
 If i want a normal like picture 2, is there some way easy?
![jolt](https://github.com/user-attachments/assets/eebe246f-9cfb-4da0-95e0-02b1debc1a11)
![bep](https://github.com/user-attachments/assets/b697ca45-aa0d-4a49-8cd8-afc2ca1077e3)

**Comment by jrouwe** [comment]
The algorithm used during `CastShape` (GJK) does not return a terribly accurate normal, you can however improve the results:

If the capsule in this picture collides at the start of the sweep and you want an accurate normal you'll need to set `ShapeCastSettings::mReturnDeepestPoint = true`, this will cost some performance but gives you a more accurate normal.

If the capsule in this picture collides somewhere between the start and end of the sweep you can try setting `ShapeCastSettings::mUseShrunkenShapeAndConvexRadius = true`.

**Reply by sinkinthesea** [reply]
Thank you very much ~ ^_^
